### PR TITLE
Add support for strings on the LHS of function definitions

### LINKED ISSUE #143: `tags.scm` should probably recognize functions with string names on the LHS
Like `"$<-.data.table"` in data.table
https://github.com/Rdatatable/data.table/blob/4d42cdc4f73084390eccc589e726519556c1f8a1/R/data.table.R#L2257-L2264

Screenshot 2024-09-13 at 3 26 59 PM

Not sure exactly what we should tag as the `@name`, _probably_ just the whole string including the string delimiters? I think if we naively drop the delimiters and use the `string_content` we could have issues in the case of `"name \" with dquote"` because you end up with this `\` in the captured text that would have to be unescaped to get `name " with dquote`. Seems a little complicated.

Then again, maybe if it works better in the 99% case with `string_content`, we should just do that and be ok with problems in the 1% case?

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #143

### CODE REVIEW

**Comment by MichaelChirico** [comment]
I _think_ the new tests are good & look like they're passing so my edits to tags.scm are WAI? My first time touching tree-sitter so please hand-hold if that's wrong 🙏

I think backticked names were already captured, but adding a test here since that's a useful edge case to keep tabs on.

I wonder if shorthand-lambdas should be included as well, but decided against it as out-of-scope for this PR for now, WDYT?

File: test/tags/function-definitions.R [hunk 1]
```diff
@@ -1,14 +1,38 @@
 fn <- function() {
-# <- definition.function  
+# <- definition.function
```
Comment (MichaelChirico) [inline test/tags/function-definitions.R:2]: oh, in-browser VS editor made these WS changes, are they important?

Comment (DavisVaughan) [reply test/tags/function-definitions.R:2]: I don't think so! I must have had whitespace trimming off by accident when I touched these last. Fine as is

**Comment by MichaelChirico** [comment]
Friendly ping! If you're swamped that's fine, just reminding before I forget about it myself :)

**Comment by DavisVaughan** [comment]
> I wonder if shorthand-lambdas should be included as well

This actually would be included because native anonymous functions are part of `function_definition`

```r
fn <- \(x) x
```
